src/AudioIO.cpp

```diff
@@ -1748,12 +1762,34 @@ AudioThread::ExitCode AudioThread::Entry()
          gAudioIO->TrackBufferExchange();
          gAudioIO->mAudioThreadShouldCallTrackBufferExchangeOnce
             .store(false, std::memory_order_relaxed);
+
+         lastState = State::eOnce;
       }
       else if( gAudioIO->mAudioThreadTrackBufferExchangeLoopRunning
          .load(std::memory_order_relaxed))
       {
-         gAudioIO->TrackBufferExchange();
+         if (lastState != State::eLoopRunning)
+         {
+            // Main thread has told us to start - acknowledge that we do
+            gAudioIO->mAudioThreadAcknowledge.store(Acknowledge::eStart,
+                                                    std::memory_order::memory_order_release);
```

Comment:
Is it all right that this store happens before TrackBufferExchange?  Yes, because the main thread only needs acknowledgment that the message was seen.

This is unlike the case with mAudioThreadShouldCallTrackBufferExchangeOnce where the store should really mean that the one-time exchange is done.

It may be worth commenting this difference.